Revised TextInput.md (#2387)
coveo/plasma: packages/website/docs/TextInput.md

[Description]: # 
[Text inputs allow to enter and edit short texts, such as names, emails or passwords.]: # 
>

> > ### Best Practices :bulb: 

Reviewer: I think we'll need to add svg or icons in the repo if we want to use them in the guidelines :thinking: 
:bulb: is not rendering in the demo.

Author: Don't bother, I will remove them, it was an hopeful test but not required for now.

Reviewer 2: real emojis should work

## Labeling

Keep labels short, preferably **under three words**.

Provide a descriptive title without action verbs. For example, write "Email" rather than "Enter your email".

## Help Text and Instructions

Help text provides extra guidance on what information to provide or how the user should fill it in, for example when there are character restrictions.

Help text should be **short, preferably on one line**.
There are two ways to provide help text:

-   Written instructions  (e.g., "Provide your business email.")
-   Example (e.g., "E.g., johndoe@acme.com")

A combination of both is also acceptable.

The label of the text input should be self-explanatory.
Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.

## Feedback and Validation

If there are character restrictons, consider indicating them in the help text. Avoid relying only on the validation message to inform the user about them.

If the user provides text that doesn't meet the requirements, the validation message should clearly indicate how to fix it.

> If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.

Reviewer 3: ```suggestion
If providing the information is optional, the text input will be automatically tagged as such, indicating that the input can be left blank.
```

## Labeling

Keep labels short, **under three words**.

Use descriptive labels without action verbs.

-   E.g. Use "Email" instead of "Write your email".

## Help text and instructions

Help text provide extra guidance on **what** information must be provided or **how** the user should fill it in (such as restricted characters).

Help text should be **short one-liner**.
There are two ways to provide help text guidance:

-   Written instructions : "Provide your business email."
-   Example given : "E.g.: johndoe@acme.com"

A combination of both is also acceptable.

The label of the text input should be self-explanatory as much as possible.
Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.

## Feedback and validation

If some characters are restricted, consider indicating it in an help text instead of relying only on the validation message to inform the user about them.

If the provided information isn't compliant with the requirements, make sure that the validation feedback clearly indicates how to fix it in the text of the validation message.

> If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.

Reviewer 3: @kpoulincoveo It's unclear to me what we mean here.
"If the information is optional" = if it's optional that the user provides text, right?
"the text input will automatically be tagged" = the placeholder contains `(optional)`? 🤔

---

## Related Components

> If your use case doesn't match these guidelines, consider using alternate component instead.

Reviewer 3: ```suggestion
If your use case doesn't match the guidelines above, consider using one of the following components instead:
```

> -   [Text Area](#/form/TextArea) when you require more than one line of characters.

Reviewer 3: @kpoulincoveo Again, just to confirm my understanding: does this mean that the user MUST provide 2+ lines, or that the information that they input MAY stand on 2+ lines (ex: two mapping rules, each on one line).

Author: It's more like a paragraph than two distinctive lines.

Reviewer 3: OK so how about:
```suggestion
-   [Text Area](#/form/TextArea) to have users enter longer text.
```

-   [Numeric Input](#/form/NumericInput) when you restrict characters to only numerical values.

> -   [Single-Select](#/form/SingleSelect) when you instead want user to select an option from a predefined list.

Reviewer 3: ```suggestion
-   [Single select](#/form/SingleSelect) to have users select an option from a list instead.
```

## Best Practices

Use text inputs when the user must enter one **short line of text.**

The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases.

> The minimum width is 320px and can be incremented by multiples of 8.

Author: I will remove it for now. thanks

> ## Labeling

Author: As I said in Checkbox. Text is too general. We'll keep labeling for now.

Keep labels short, **under three words**.

Use descriptive labels without action verbs.

-   E.g. Use "Email" instead of "Write your email".

## Help text and instructions

Help text provide extra guidance on **what** information must be provided or **how** the user should fill it in (such as restricted characters).

Help text should be **short one-liner**.
There are two ways to provide help text guidance:

-   Written instructions : "Provide your business email."
-   Example given : "E.g.: johndoe@acme.com"

A combination of both is also acceptable.

The label of the text input should be self-explanatory as much as possible.

> Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.

Author: What I meant here is that title and description should only be added when additional guidance is critical to the user experience. 

Revision changes:
Add a title and description when providing additional guidance is critical for the user to understand his task.

> Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.

Reviewer 3: Ah, I see, thanks!
```suggestion
Add a title and description when providing additional guidance is critical for the user to understand why the information is required or how the information will be used.
```

[Description]: # 
[Text inputs allow to enter and edit short texts, such as names, emails or passwords.]: # 

### Best Practices
Use text inputs when the user must enter one **short line of text.**

The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases. 
The minimum width is 320px and can be incremented by multiples of 8.

---
### Labeling

> Keep labels short, under three words. 

Author: Suggestion:  Keep labels short, **under three words**.

> Use descriptive labels. Avoid action verbs. (E.g. Use "Email" instead of "Write your email")

Author: Suggestions: 
Use descriptive labels without action verbs. 
- E.g. Use "Email" ...

---
### Help text and instructions
Help text provide extra guidance on **what** information must be provided or **how** the user should fill it in (such as restricted characters).

Help text should be **short one-liner**. 
There are two ways to provide help text guidance: 
 
- Written instructions : "Provide your business email." 
- Example given: "E.g.: johndoe@acme.com" 

A combination of both is also acceptable. 

> Text input should be self-explanatory as much as possible.

Author: Suggestion: 
The label of the text input should be self-explanatory as much as possible.

Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.

---
### Feedback and validation
If some characters are restricted, consider indicating it in an help text instead of relying only on the validation message to inform the user about them.

If the provided information isn't compliant with the requirements, make sure that the validation feedback clearly indicates how to fix it in the text of the validation message.

If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.

---
### Related Components
If your use case doesn't match these guidelines, consider using alternate component instead. 
 
- [Text Area](#/components/TextArea) when you require more than one line of characters.
- [Numeric Input](#/components/NumericInput) when you restrict characters to only numerical values.

> - [Single Select](#/input/SingleSelect) when you instead want user to select an option from a predefined list.

Author: Correction:
#/input/TextArea
#/input/NumericInput

[Description]: # 
[Text inputs allow to enter and edit short texts, such as names, emails or passwords.]: # 

## Best Practices
Use text inputs when the user must enter one **short line of text.**

> The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases. 

Reviewer 4: [Nitpick] if it's best practice I don't think you have to specify that this guideline apply in most case. 
```suggestion
The width of the text input should be proportional to the expected user input and large enough to display the entire text. 
```
